eql/parser.py

```diff
@@ -880,11 +889,14 @@ def function_call(self, node, prev_node=None, prev_arg=None):
 
             # Get all of the arguments first, because they may depend on others
             # and we need to pull out all of the variables
+            var_type = None
+            var_schema = None
+
             for idx, arg_source in enumerate(nodes):
                 if idx in variables:
                     if arg_source.data == "base_field":
                         variable_name = self.visit(arg_source["name"])
-                        self._add_variable(variable_name)
+                        self._add_variable(variable_name, var_type or TypeHint.Unknown, var_schema)
```

Comment:
Do we have to add an special case for `arg_source.data  == 'varpath'`?

Reply:
What do you mean? Can you give an example?

Reply:
![Screenshot 2023-07-21 at 5 01 36 PM](https://github.com/endgameinc/eql/assets/1636709/8b5223e5-b5cc-4a29-9c2f-ebac235d74b1)

Right now the logic is erroring when arg_source.data is `varpath`. Is that expected or should we add another else if statement for the `$var`.